src/services/services.ts

```diff
@@ -4639,7 +4639,16 @@ namespace ts {
             // to jump to the implementation directly.
             if (symbol.flags & SymbolFlags.Alias) {
                 const declaration = symbol.declarations[0];
-                if (node.kind === SyntaxKind.Identifier && node.parent === declaration) {
+
+                // Go to the original declaration for cases:
+                //
+                //   (1) when the aliased symbol was declared in the location(parent).
+                //   (2) when the aliased symbol is originating from a named import.
+                //
+                if (node.kind === SyntaxKind.Identifier &&

```

Comment:
@vladima we only want to go to the ultimate location on named import and not all the time? Otherwise, I think this is not needed.

Do you expect to go to the ultimate location on this cases as well:
```typescript
import a from './a';
let test = /*1*/a;
```

```typescript
import a = require('./a');
let test = /*1*/a;
```
Maybe just add another rule that on explicit alias(an alias you can put any name on), you go to the ultimate location?

So for instance:

```typescript
import { a as b } from './a';
let c = b; // this will not go to ultimate.
```
```typescript
import { a } from './a';
let c = a; // this will go to ultimate.
```
Then we got more consistency.